cfme/tests/physical_infrastructure/test_physical_server_details.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.utils import testgen
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils.wait import wait_for
+
+pytestmark = [pytest.mark.tier(3)]
+
+pytest_generate_tests = testgen.generate([LenovoProvider], scope="module")
+
+
+@pytest.fixture(scope="module")
+def physical_server(appliance, provider):
+    # Create and then wait for the provider to be added
+    provider.create()
+    wait_for(
+        lambda: provider.get_detail('Relationships', 'Physical Servers') != '0',
+        fail_func=provider.refresh_provider_relationships_ui,
+        num_sec=60,
+        delay=5
+    )
+
+    # Get and return the first physical server
+    physical_servers = appliance.collections.physical_servers.all(provider)
+    physical_server = physical_servers[0]
+    yield physical_server
+
+    # Clean up the provider when finished
+    provider.delete(cancel=False)
```

Comment:
No need for this, using the existing setup_provider fixtures will cover removing the provider when its no longer necessary.